src/main/java/com/iexec/worker/feign/CustomCoreFeignClient.java

```diff
@@ -79,10 +80,10 @@ public String getCoreVersion() {
     }
 
     public String ping() {
-        Map<String, Object> arguments = new HashMap<>();
-        arguments.put(JWTOKEN, loginService.getToken());
-        HttpCall<String> httpCall = args -> coreClient.ping((String) args.get(JWTOKEN));
-        ResponseEntity<String> response = makeHttpCall(httpCall, arguments, "ping");
+        ResponseEntity<String> response = coreClient.ping(loginService.getToken());
+        if (response.getStatusCode() == HttpStatus.UNAUTHORIZED) {
+            loginService.login();
```

Comment:
Shouldn't we reping here?

Reply:
The strategy here seems to be waiting for the next ping to be triggered. It seems fine to me as this, don't you think so? In the case we'd like to ping again right after the login, what should we do if we get another unauthorized error? Should we log in and ping another time?

Reply:
Yes, that seems fair, maybe we can add a note on that and a return after login

Reply:
I tried another implementation, see 53bc7e3